packages/tldraw/src/lib/shapes/arrow/arrowLabel.ts

```diff
@@ -309,6 +306,47 @@ export function getArrowLabelPosition(editor: Editor, shape: TLArrowShape) {
 	return { box: Box.FromCenter(labelCenter, labelSize), debugGeom }
 }
 
+function getClampedPosition(
+	editor: Editor,
+	shape: TLArrowShape,
+	range: { start: number; end: number },
+	arrowheadInfo: ArrowheadInfo
+) {
+	const { hasEndArrowhead, hasEndBinding, hasStartBinding, hasStartArrowhead } = arrowheadInfo
+	const arrowLength = getArrowLength(editor, shape)
+	let clampedPosition = clamp(
+		shape.props.labelPosition,
+		hasStartArrowhead || hasStartBinding ? range.start : 0,
+		hasEndArrowhead || hasEndBinding ? range.end : 1
+	)
+	const minLength = 100
+	const maxLength = 1500
+	const minSnapDistance = 0.03
+	const maxSnapDistance = 0.005
+
+	let snapDistance = minSnapDistance
```

Comment:
i think we can simplify this all. you should be able to do just this:
`const snapDistance = Math.min(0.02, 500 / arrowLength * 0.02)`

basically, we want it to make out at 0.02 but as the arrow grows longer we want it to go smaller and tighter to that midpoint